hudi-flink/src/main/java/org/apache/hudi/util/StreamerUtil.java

```diff
@@ -332,7 +331,7 @@ public static String getTablePath(FileSystem fs, Path[] userProvidedPaths) throw
   public static boolean needsScheduleCompaction(Configuration conf) {
     return conf.getString(FlinkOptions.TABLE_TYPE)
         .toUpperCase(Locale.ROOT)
-        .equals(HoodieTableType.MERGE_ON_READ.name())
+        .equals(FlinkOptions.TABLE_TYPE_MERGE_ON_READ)
```

Comment:
I still think the old one is better. We should have the only fact source. If we define a new one, it would bring more cost of maintenance.

Reply:
Not true, because `HoodieTableType` should never change.

Reply:
I am not sure if a definition would be changed. But if we have a fact source, anywhere we reference it, it's enough. IMO, it's not necessary to define it more than once. Otherwise, the others would define it in `java` or `spark` module? Because, we allowed this behavior in `flink` module. It will add the complexity and the cost of understanding.

Reply:
IMO, it's different. Spark referenced the `HoodieTableType.COPY_ON_WRITE.name`, but you used a string literal. What I mean about the word `define`, is to use a literal directly. In the future, if we change the value of `HoodieTableType `, in `DataSourceOptions ` we do not need to search the string literal, and we can control all the change points.

Reply:
`HoodieTableType` is an enumeration, but in many cases we need a string constant, not a method call like `HoodieTableType.name()`, such as the Junit5 parameterized test.

BTW, can we not block this PR because these two options are now introduced in this PR though.

Reply:
We did not block the PR because of the reason for this divergence.

Reply:
We can do that.